### ISSUE TITLE
fix(reranker): l'échelle de scores a changé avec le passage sur Jetson — défaut 0.0, min_score et docstrings périmés

### ISSUE THREAD

**Issue by didlawowo** [didlawowo/rag-doc#5]
## Contexte — le *pourquoi*

Le reranking a changé de moteur le 2026-08-25 : `reranker.dc-tech.work` (nœud nvidia, éteint)
→ Jetson `192.168.1.42:8084`, servi par llama.cpp avec `bge-reranker-v2-m3-Q8_0.gguf`.

Le nouveau moteur ne rend **pas la même échelle de scores**. Mesuré sur la même paire
question/passages :

| Moteur | Passage pertinent | Bruit | Échelle |
|---|---|---|---|
| Ancien (`reranker.dc-tech.work`) | `0.990` | `0.000028` | sigmoïde, bornée `[0,1]` |
| Jetson (`bge-reranker-v2-m3` / llama.cpp) | `0.131` | `-11.034` | **logits bruts, non bornés, négatifs possibles** |

Commande de reproduction :

```bash
curl -s -X POST http://192.168.1.42:8084/v1/rerank -H 'Content-Type: application/json' -d '{"model":"reranker","query":"contrat de bail","documents":["Le bail commercial est signe pour neuf ans.","La recette des crepes demande trois oeufs."]}'
```

Le **tri reste correct** — la transformation est monotone, l'ordre est le même. Rien n'est
cassé aujourd'hui. Mais trois endroits du code raisonnent sur l'ancienne échelle, et
deviennent faux ou trompeurs.

---

### 1. `retrieval/reranker.py:48` — le défaut `0.0` n'est plus le pire score

```python
scores = [0.0] * len(documents)
for item in results:
    index = item.get("index")
    if isinstance(index, int) and 0 <= index < len(scores):
        scores[index] = float(item.get("relevance_score", 0.0))
```

Un document absent de la réponse du serveur garde `0.0`. En échelle sigmoïde, `0.0` était le
**pire score possible** : le document non noté tombait en fin de liste, ce qui est le
comportement voulu. En logits, `0.0` est un score **moyen-haut** : sur la mesure ci-dessus, il
passerait devant le passage pertinent (`0.131` étant à peine au-dessus) et **très largement**
devant le bruit (`-11.034`).

Autrement dit : un document que le reranker n'a pas noté remonte désormais devant des
documents qu'il a explicitement jugés hors sujet.

Correction : `float("-inf")` comme sentinelle, ou mieux, ne pas fabriquer de score du tout et
conserver le rang RRF d'origine pour les documents non notés. La seconde option est plus sûre
— elle n'invente aucune information.

> Ce chemin est atteignable en pratique : `ocr.py:113-115` documente déjà des réponses
> HTTP 200 au corps incomplet côté fournisseur. Rien ne garantit que `results` couvre
> systématiquement tous les documents envoyés.

### 2. `retrieval/reranker.py:78-79` — `min_score` devient un piège

```python
if settings.reranker.min_score > 0:
    kept = [h for h in rescored if h[1] >= settings.reranker.min_score]
    rescored = kept or rescored[:1]
```

Le défaut est `0.0` (`config.py:103`), donc **le seuil n'est pas appliqué aujourd'hui** et rien
ne dysfonctionne. Mais le champ reste offert, et n'importe quelle valeur qui avait du sens en
sigmoïde (`0.5`, `0.8`…) écarte maintenant la quasi-totalité des passages — y compris les bons,
qui plafonnent ici à `0.131`. Le garde-fou `kept or rescored[:1]` limite les dégâts à « un seul
résultat » au lieu de zéro, ce qui reste une recherche muette en pratique.

Deux options :

- **(a)** normaliser les scores par une sigmoïde à la réception, ce qui restaure l'échelle
  `[0,1]` et rend `min_score` de nouveau interprétable — au prix d'une transformation qui
  n'est pas celle du modèle d'origine ;
- **(b)** documenter que `min_score` est **dépendant du backend**, et le valider au démarrage
  (refuser `min_score > 0` quand le backend rend des logits).

L'option (a) est probablement la bonne : elle rend le champ utilisable et supprime aussi le
problème du point 1 (en sigmoïde, `0.0` redevient le pire score). À trancher.

### 3. Docstrings et défauts périmés

| Réf. | Contenu périmé | Réalité |
|---|---|---|
| `retrieval/reranker.py:6-7` | « mesuré sur le reranker local : 0.990 / 0.000028, quatre ordres de grandeur d'écart » | Mesure de l'**ancien** moteur. Le chiffre est vrai historiquement mais décrit un moteur qui n'existe plus |
| `config.py:99-102` | « son contexte est court (1024 tokens) », d'où `max_chars=2000` | `bge-reranker-v2-m3` annonce `n_ctx_train: 8192`. La troncature à 2000 caractères est désormais **huit fois** plus agressive que nécessaire — des passages sont amputés avant notation |
| `config.py:95` | défaut `https://reranker.dc-tech.work/v1` | **Hôte mort**, ne résout plus en DNS |
| `config.py:96` | défaut `model="reranker"` | Le Jetson sert `bge-reranker-v2-m3-Q8_0.gguf` (l'alias `reranker` fonctionne toutefois — llama.cpp ignore le nom) |

Le point sur `max_chars` n'est pas cosmétique : c'est une perte de qualité silencieuse. Un
passage de 6000 caractères est coupé à 2000 avant d'être noté, alors que le modèle pourrait
l'ingérer entier.

---

## Tests unitaires à écrire

`tests/test_reranker.py` existe déjà — à compléter. Pas de service externe.

| Test | Couvre |
|---|---|
| `test_rerank_should_rank_unscored_document_last_when_server_omits_it` | point 1 — le cœur de l'issue |
| `test_rerank_should_preserve_rrf_order_for_unscored_documents` | si l'option « pas de score inventé » est retenue |
| `test_rerank_should_order_correctly_when_scores_are_negative_logits` | non-régression sur l'échelle réelle du Jetson |
| `test_rerank_should_keep_one_hit_when_min_score_excludes_all` | garde-fou `kept or rescored[:1]` existant |
| `test_normalize_should_map_logits_to_unit_interval` | si l'option (a) sigmoïde est retenue |

Valeurs de test à reprendre telles quelles : `0.131` (pertinent) et `-11.034` (bruit), mesurées
sur le Jetson — elles ancrent le test dans le comportement réel plutôt qu'inventé.

---

## Décision à trancher

**Option (a) sigmoïde à la réception, ou option (b) `min_score` dépendant du backend ?**

Le reste de l'issue en découle : (a) règle les points 1 et 2 d'un coup, (b) demande une
sentinelle `-inf` au point 1 et une validation au démarrage au point 2.

---

## Vérification

**Côté agent (automatisé)**

- [ ] `task test` — les tests ci-dessus passent
- [ ] `task lint` / `task fmt:check` / `task typecheck`

**Côté utilisateur (manuel)**

- [ ] Poser une question à l'agent dont la bonne réponse est en 5ᵉ position RRF, vérifier
      qu'elle remonte
- [ ] Après relèvement de `max_chars` : comparer les passages retenus sur un document à
      chunks longs, avant/après

## Risques

| # | Risque | Mitigation |
|---|---|---|
| Q1 | Une sigmoïde ajoutée à la main n'est pas la calibration du modèle | Elle sert au seuillage et au tri, pas à une interprétation probabiliste — à écrire dans le docstring |
| R1 | Relever `max_chars` allonge les requêtes de reranking | 30 candidats × 8000 caractères reste modeste ; à mesurer sur le Jetson, qui est moins puissant que l'ancien nœud GPU |
| Q2 | Un futur changement de backend recasse l'échelle | C'est précisément ce que le test sur logits négatifs verrouille |

## Rollback

Aucune donnée indexée n'est concernée — le reranking s'applique **à la lecture**, pas à
l'ingestion. Revenir en arrière ne demande aucune ré-indexation.
